src/data_structures.rs

```diff
@@ -26,7 +30,6 @@ impl Stats{
 #[derive(PartialEq, Eq, Hash, Clone, Copy, Debug)]
 pub struct Segment{
```

Comment:
So I think it'd be good to change this so Segment is agnostic to whatever the underlying thing is.

e.g. basically the Segment is type T where T implements Clone + Hash + Eq + PartialEq + Segment

where Pub Trait Segment {
fn segment_id(&self) -> SegmentID;
fn page_size(&self) -> PageSizeEnum;
}

then the hashmap you've defined is also generic over this type T
so it's more like
pub struct ZeroCopyCache<T> where T: all those traits above
and then it contains a Hasmap<T, Stats>


But we can also defer this change for later